### PR TITLE
POM fixes.

### PR DESCRIPTION
**Pull request by Sxtanna**
[Pull requests]: https://github.com/PlaceholderAPI/PlaceholderAPI/pulls
[contributing file]: https://github.com/PlaceholderAPI/PlaceholderAPI/tree/master/.github/CONTRIBUTING.md

## Please read
Please make sure you checked the following:
- You checked the [Pull requests] page for any upcoming changes.
- You documented any public code that the end-user might use.
- You followed the [contributing file] 

### Type

- [x] Internal change (Doesn't affect end-user).
- [ ] External change (Does affect end-user).
- [ ] Other: __________ 

Fixed unnecessary changed to the pom

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -33,10 +35,16 @@
...
+            <artifactId>spigot-api</artifactId>
             <version>1.16.1-R0.1-SNAPSHOT</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+            <version>2.13.3</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (Andre601) [inline pom.xml:47]: Why is that added?

Comment (Sxtanna) [reply pom.xml:47]: I don't know if you noticed, but the last 1600+ merged PR contained code that referenced classes provided by internal spigot dependencies.

This is why they changed the dependency from `spigot-api` to `spigot`, rendering the entire project unworkable unless you had the server jar built in your local repo.

I reverted this, and added a provided dependency for what those classes referenced.
